File: engine/src/main/java/com/example/mypixel/model/node/StringNode.java
package com.example.mypixel.model.node;

import com.example.mypixel.model.Parameter;
import com.example.mypixel.model.ParameterType;
import com.example.mypixel.model.ParamsMap;
import com.example.mypixel.model.Widget;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.lang.NonNull;

import java.util.Map;

@MyPixelNode("String")
public class StringNode extends Node {

    @JsonCreator
    public StringNode(
            @JsonProperty("id") @NonNull Long id,
            @JsonProperty("type") @NonNull String type,
            @JsonProperty("inputs") Map<String, Object> inputs) {
        super(id, type, inputs);
    }

    @Override
    public Map<String, Parameter> getInputTypes() {
        return ParamsMap.of(
                "input", Parameter.builder()
                        .type(ParameterType.STRING)
                        .required(true)
                        .widget(Widget.INPUT)
                        .build()
        );
    }

    @Override
    public Map<String, Object> getDefaultInputs() {
        return ParamsMap.of(
                "input", ""
        );
    }

    @Override
    public Map<String, Parameter> getOutputTypes() {
        return ParamsMap.of("output", Parameter.required(ParameterType.STRING));
    }

    @Override
    public Map<String, String> getDisplayInfo() {
        return ParamsMap.of(
                "category", "Types",
                "description", "String",
                "color", "#AED581",
                "icon", "StringIcon"
        );
    }

    @Override
    public Map<String, Object> exec() {
        Map<String, Object> outputs;
        String value = (String) inputs.get("input");
        outputs = Map.of("output", value);
        return outputs;
    }

    @Override
    public void validate() {

    }
}

File: node/nodes/string_node.py
from typing import Dict, Any
from node import Node, register_node_class


@register_node_class
class StringNode(Node):

    node_type = "String"

    def get_input_types(self) -> Dict[str, Dict[str, Any]]:
        return {
            "input": {
                "type": "STRING",
                "required": True,
                "widget": "INPUT",
                "default": ""
            }
        }

    def get_output_types(self) -> Dict[str, Dict[str, Any]]:
        return {
            "output": {
                "type": "STRING",
                "required": True
            }
        }

    def get_display_info(self) -> Dict[str, str]:
        return {
            "category": "Types",
            "description": "String",
            "color": "#AED581",
            "icon": "StringIcon"
        }

    def exec(self, value: str) -> Dict[str, Any]:
        return {"output": value}

    def validate(self, value: str) -> None:
        pass